File: src/Bk.APG.UI/src/api/GeneralElectionCommitteeFilterParameters.ts
import {CommitteeType} from './CommitteeType';
import {Department} from './Department';
import {Level} from './Level';
import {Office} from './Office';

export interface GeneralElectionCommitteeFilterParameters {
    freeText?: string;
    levels?: Level['id'][];
    departments?: Department['id'][];
    offices?: Office['id'][];
    committeeTypes?: CommitteeType['id'][];
    isMarketOrientated?: boolean[];
    hasSupervisionDuty?: boolean[];
    isNew?: boolean[];
    vacancies?: boolean[];
    statusProposal?: boolean[];
    modifiedBy?: string;
}

File: src/Bk.APG.Business/Models/GeneralElectionCommitteeFilterParameters.cs
namespace Bk.APG.Business.Models;

public class GeneralElectionCommitteeFilterParameters
{
    public string? FreeText { get; init; }
    public IEnumerable<Guid>? LevelIds { get; init; }
    public IEnumerable<Guid>? DepartmentIds { get; init; }
    public IEnumerable<Guid>? OfficeIds { get; init; }
    public IEnumerable<Guid>? CommitteeTypeIds { get; init; }
    public IEnumerable<bool>? IsMarketOrientated { get; init; }
    public IEnumerable<bool>? HasSupervisionDuty { get; init; }
    public IEnumerable<bool>? IsNew { get; init; }
    public IEnumerable<bool>? Vacancies { get; init; }
    public IEnumerable<bool>? StatusProposal { get; init; }
    public IEnumerable<Guid>? CommitteeIds { get; set; }
}
